testing_utilities.go

```diff
@@ -0,0 +1,359 @@
+
+			switch intSize {
+			case ion.Int32, ion.Int64:
+				int1, err := reader1.Int64Value()
+				assert.NoError(t, err, "Something went wrong executing reader1.Int64Value()")
+
+				int2, err := reader2.Int64Value()
+				assert.NoError(t, err, "Something went wrong executing reader2.Int64Value()")
+
+				assert.Equal(t, int1, int2, "Expected int values to match")
+			case ion.Uint64:
+				uint1, err := reader1.Uint64Value()
+				assert.NoError(t, err, "Something went wrong executing reader1.Uint64Value()")
+
+				uint2, err := reader2.Uint64Value()
+				assert.NoError(t, err, "Something went wrong executing reader2.Uint64Value()")
+
+				assert.Equal(t, uint1, uint2, "Expected uint values to match")
+			case ion.BigInt:
+				bigInt1, err := reader1.BigIntValue()
+				assert.NoError(t, err, "Something went wrong executing reader1.BigIntValue()")
+
+				bigInt2, err := reader2.BigIntValue()
+				assert.NoError(t, err, "Something went wrong executing reader2.BigIntValue()")
+
+				assert.Equal(t, bigInt1, bigInt2, "Expected big int values to match")
+			default:
+				t.Error("Expected intSize to be one of Int32, Int64, Uint64, or BigInt")
+			}
+		}
+	case ion.FloatType:
+		if !isNull {
+			float1, err := reader1.FloatValue()
+			assert.NoError(t, err, "Something went wrong executing reader1.FloatValue()")
+
+			float2, err := reader2.FloatValue()
+			assert.NoError(t, err, "Something went wrong executing reader2.FloatValue()")
+
+			if math.IsNaN(float1) && math.IsNaN(float2) {
+				assert.Equal(t, float1, float2, "Expected NaN float values to match")
+			} else if math.IsNaN(float1) || math.IsNaN(float2) {
+				assert.NotEqual(t, float1, float2, "Expected float values to differ")
+			} else {
+				assert.Equal(t, float1, float2, "Expected float values to match")
+			}
+		}
+	case ion.DecimalType:
+		if !isNull {
+			decimal1, err := reader1.DecimalValue()
+			assert.NoError(t, err, "Something went wrong executing reader1.DecimalValue()")
+
+			decimal2, err := reader2.DecimalValue()
+			assert.NoError(t, err, "Something went wrong executing reader2.DecimalValue()")
+
+			decimalStrictEquals(t, decimal1, decimal2)
+		}
+	case ion.TimestampType:
+		if !isNull {
+			timestamp1, err := reader1.TimeValue()
+			assert.NoError(t, err, "Something went wrong executing reader1.TimeValue()")
+
+			timestamp2, err := reader2.TimeValue()
+			assert.NoError(t, err, "Something went wrong executing reader2.TimeValue()")
+
+			assert.Equal(t, timestamp1, timestamp2, "Expected timestamp values to match")
+		}
+	case ion.StringType:
+		str1, err := reader1.StringValue()
+		assert.NoError(t, err, "Something went wrong executing reader1.StringValue()")
+
+		str2, err := reader2.StringValue()
+		assert.NoError(t, err, "Something went wrong executing reader2.StringValue()")
+
+		assert.Equal(t, str1, str2, "Expected string values to match")
+	case ion.SymbolType:
+		// TODO: Add SymbolToken logic here once SymbolTokens are available
```

Comment:
Till we get the TODO should we just fail for now? or perform the test if we have a non SID (i.e. we have actual text for the symbol)?

Reply:
Test fails for now.